android-components/components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/CustomTabWindowFeature.kt

```diff
@@ -4,73 +4,27 @@
 
 package mozilla.components.feature.customtabs
 
-import android.app.Activity
-import android.content.ActivityNotFoundException
-import android.net.Uri
-import androidx.annotation.VisibleForTesting
-import androidx.annotation.VisibleForTesting.Companion.PRIVATE
-import androidx.browser.customtabs.CustomTabColorSchemeParams
-import androidx.browser.customtabs.CustomTabsIntent
-import androidx.core.net.toUri
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.cancel
 import kotlinx.coroutines.flow.distinctUntilChangedBy
 import kotlinx.coroutines.flow.mapNotNull
 import mozilla.components.browser.state.action.ContentAction
 import mozilla.components.browser.state.selector.findCustomTab
-import mozilla.components.browser.state.state.CustomTabConfig
 import mozilla.components.browser.state.store.BrowserStore
 import mozilla.components.concept.engine.window.WindowRequest
 import mozilla.components.lib.state.ext.flowScoped
 import mozilla.components.support.base.feature.LifecycleAwareFeature
 
-const val SHORTCUT_CATEGORY = "mozilla.components.pwa.category.SHORTCUT"
-
 /**
  * Feature implementation for handling window requests by opening custom tabs.
  */
 class CustomTabWindowFeature(
-    private val activity: Activity,
     private val store: BrowserStore,
     private val sessionId: String,
-    internal val onLaunchUrlFallback: (Uri) -> Unit,
 ) : LifecycleAwareFeature {
 
     private var scope: CoroutineScope? = null
 
-    /**
-     * Transform a [CustomTabConfig] into a [CustomTabsIntent] that creates a
-     * new custom tab with the same styling and layout
-     */
-    @Suppress("ComplexMethod")
-    @VisibleForTesting(otherwise = PRIVATE)
-    internal fun configToIntent(config: CustomTabConfig?): CustomTabsIntent {
```

Comment:
Since this PR largely involves a big function deletion, I need some help understanding not the bug within the experience but rather within the code.

So was this function essentially trying to spin off another custom tab from the presently-open existing custom tab? So then putting it all together: 
- Users tries tapping on certain links
- Fenix tries to spin-off another custom tab when it's already in a custom tab
- Nothing happens

?

Reply:
Sure.  When a new window request happens, (could be because the site needed to open a new window, or the user taps on something that triggers a new window) we try to open that window in a new custom tab.  However, since custom tab doesn't have multiple tabs, the current way is we trigger a brand new separate custom tab instance.  Once the user is done with that custom tab, they close it and get back to the original custom tab (that sends the new window request).

But because we're triggering a new custom tab instance, a lot of things will fail.  In this particular case, a blob:// url is not something we support coming from another application (Not declared in Android Manifest), and since we consider the new custom tab request external (a new intent), we can't load it.  (similar if we get a javascript:// url etc).

Reply:
But all that said, the original behaviour of opening a new separate custom tab does not match what Chrome does.  If you try this with Chrome set as default browser ```adb shell am start -a android.intent.action.VIEW -d https://jsfiddle.net/actyp5gr/ -e android.support.customtabs.extra.SESSION true```, you'll see that Chrome opens the new window inside the same custom tab.

So in a way, this change aligns us with Chrome's behaviour and at the same time fixed the problem that if we're opening a new url in custom tab, we shouldn't be opening it in a new window because custom tab should always be one window.

Reply:
Gotcha gotcha. So we DO allow for, let's call it, "stacked" custom tabs, but because in this situation, there are URLs we don't support when coming from certain origins, causing the URL to just not open or do anything? Do we have a guarantee then that this behavior change will only be for custom tabs then and not the case you outlined above of one of the unsupported URLs from an external app? (The answer seems obvious but just making sure 😅)

Matching the behavior from Chrome for now makes sense to me. I'm admittedly still getting a handle on custom tabs as a whole, since there's no analog on iOS, so appreciate you walking me through all of this.

Reply:
This will only affect custom tabs.  However with Jon's comment below, I'll have to make sure we're not breaking anything that needs to have a separate tab.